app/models/user.rb

```diff
@@ -206,7 +207,17 @@ def self.authenticator(username = nil)
   end
 
   def self.authenticate(username, password, request = nil, options = {})
-    authenticator(username).authenticate(username, password, request, options)
+    user = authenticator(username).authenticate(username, password, request, options)
+    user.try(:link_user_to_session, request)
+
+    user
+  end
+
+  def link_user_to_session(request)
+    return unless request
+    
+    session_id = request.session_options[:id]
+    sessions << Session.find_or_create_by(:session_id => session_id)
```

Comment:
Wonder if we can commit all of this user.rb file in a separate PR?  It feels like the framework for deleting sessions is standalone and solid.

Reply:
Yeah, I was building this on top of Gregg's PR because I could then test it, but you have a good point that the changes themselves don't need it and I can just test with the change cherry picked locally.

I will rebase so we can do just that.  Thanks.